cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go

```diff
@@ -896,3 +893,25 @@ func TestValidateDiscoveryKubeConfigPath(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateSocketPath(t *testing.T) {
+	var tests = []struct {
+		name           string
+		sp             string
+		errorsExpected bool
+	}{
+		{name: "valid path", sp: "/some/path", errorsExpected: false},
+		{name: "valid socket url", sp: kubeadmapiv1beta2.DefaultUrlScheme + "://" + "/some/path", errorsExpected: false},
+		{name: "unsupported url scheme", sp: "bla:///some/path", errorsExpected: true},
+		{name: "unparseable url", sp: ":::", errorsExpected: true},
+		{name: "invalid CRISocket (path is not absolute)", sp: "some/path", errorsExpected: true},
+		{name: "empty CRISocket (path is not absolute)", sp: "", errorsExpected: true},
+	}
+	for _, tc := range tests {
+		actual := ValidateSocketPath(tc.sp, field.NewPath("socketPath"))
+		actualErrors := len(actual) > 0
+		if actualErrors != tc.errorsExpected {
+			t.Errorf("error: test name: %q, socket path: %q\n\texpected: %t\n\t  actual: %t", tc.name, tc.sp, tc.errorsExpected, actualErrors)
```

Comment:
ideally validation_test.go should use sub-tests everywhere, see here for an example:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/init_test.go#L153

this removes the need to pass tc.name in log calls (e.g. t.Errorf(...)).